api/src/pcapi/connectors/acceslibre.py

```diff
@@ -0,0 +1,148 @@
+            return erp["uuid"]
+    return None
+
+
+class AccesLibreApiException(Exception):
+    pass
+
+
+class BaseBackend:
+    def find_venue_at_accessibility_provider(
+        self,
+        name: str,
+        public_name: str | None = None,
+        siret: str | None = None,
+        ban_id: str | None = None,
+        city: str | None = None,
+        postal_code: str | None = None,
+    ) -> str | None:
+        raise NotImplementedError()
+
+
+class TestingBackend(BaseBackend):
+    def find_venue_at_accessibility_provider(
+        self,
+        name: str,
+        public_name: str | None = None,
+        siret: str | None = None,
+        ban_id: str | None = None,
+        city: str | None = None,
+        postal_code: str | None = None,
+    ) -> str | None:
+        return "11111111-2222-3333-4444-5555555555555"
+
+
+class AcceslibreBackend(BaseBackend):
+    def _send_request(
+        self,
+        query_params: dict[str, str],
+    ) -> dict:
+        api_key = settings.ACCESLIBRE_API_KEY
+        url = settings.ACCESLIBRE_API_URL
+        headers = {"Authorization": f"Api-Key {api_key}"}
+        try:
+            response = requests.get(url, headers=headers, params=query_params, timeout=ACCESLIBRE_REQUEST_TIMEOUT)
+        except requests.exceptions.RequestException:
+            raise AccesLibreApiException(
+                f"Error connecting AccesLibre API for {url} and query parameters: {query_params}"
+            )
+        try:
+            return response.json()
+        except json.JSONDecodeError:
+            logger.error(
+                "Got non-JSON or malformed JSON response from AccesLibre",
+                extra={"url": response.url, "response": response.content},
+            )
+            raise AccesLibreApiException(f"Non-JSON response from AccesLibre API for {response.url}")
+
+    def find_venue_at_accessibility_provider(
+        self,
+        name: str,
+        public_name: str | None = None,
+        siret: str | None = None,
+        ban_id: str | None = None,
+        city: str | None = None,
+        postal_code: str | None = None,
+    ) -> str | None:
+        uuid = None
+        search_criteria = [
+            {"siret": siret},
+            {"ban_id": ban_id},
+            {
+                "q": slugify(name),
+                "commune": city,
+                "code_postal": postal_code,
+                "page_size": REQUEST_PAGE_SIZE,
+            },
+        ]
+
+        for criterion in search_criteria:
+            if isinstance(criterion, dict) and all(v is not None for v in criterion.values()):
+                response = self._send_request(query_params=criterion)
+                if response["count"] and (
+                    uuid := match_by_name(results_at_provider=response["results"], name=name, public_name=public_name)
+                ):
+                    return uuid
+        return uuid
```

Comment:
[détail] Si on sort de la boucle `for`, c'est qu'on a rien trouvé. On peut donc juste `return None`. (Ça évite de définir la variable `uuid` en tout début de fonction.)